README.md

```diff
@@ -1,2 +1,133 @@
-# sparc-request-kubernetes
-Informational repostiory for building and running SPARCRequest for Kubernetes
+# SPARCRequest - Kubernetes
+
+This reference project can be used to inform how your group could run SPARCRequest in a Kubernetes environment. The following is representative how OCTRI chose to implement SPARCRequest.
+
+## Considerations for running SPARCRequest in Kubernetes
+
+### SPARCRequest Customization
+
+By using a second image we are able to modify our instance of SPARCRequest without needing to maintain a separate branch in source control.
+
+See the Image section below for a description how we build and manage our SPARCRequest instance.
+
+### Deployments
+
+We have two deployment workloads defined; one for the main application and a second for the delayed job that performs background tasks like sending emails.
+
+#### Main Deployment
+
+The main SPARCRequest application generally runs well inside Kubernetes, a single process Puma web server running the rails app. As seen in the [`deployment`](./k8s/deployment.yaml) manifest, we specify the resources, configuration, liveness, initContainers and any volumes required for persistence.
+
+The `initContainers` are used to perform database migrations and asset compilation.
+
+The volumes persist the things like attachments. We use a NFS  `PersistentVolume` to preserve these files beyond the lifespan of a pod.
+
+The `livenessProbe` informs the scheduler when the new pod is up and that it can terminate the old pod. This is important because the database migrations and asset compilation can take a bit of time and this reduces the time the application is unavailable.
+
+#### Delayed Job
+
+The challenging part of running SPARCRequest in a Kubernetes environment is running the Delayed Job as this is a separate process from SPARCRequest. To accommodate this process we run it as a [`Deployment`](./k8s/deployment_delayed_job.yaml) running the following command:
+
+```bash
+rails jobs:work
+```
+
+This can be seen in the [`deployment_delayed_job`](./k8s/deployment_delayed_job.yaml#L20) manifest. There is no reason it could not be a container in the same pod. We chose to run it as a separate pod to assist with troubleshooting and to isolate the main application in case the delayed job runs amok. This also means that we can restart it independently of the main application.
```

Comment:
I'm glad that you mentioned that this was a deliberate decision on our part. This could have been accomplished several ways, and this happened to be the best solution for our environment and containerization practices.